sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifierSerializer.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.common;
+
+import java.util.Objects;
+
+class CommunicationIdentifierSerializer {
+    public static CommunicationIdentifier deserialize(CommunicationIdentifierModel identifier) {
+        String id = identifier.getId();
+        CommunicationIdentifierKind kind = identifier.getKind();
+
+        if (kind == CommunicationIdentifierKind.COMMUNICATION_USER) {
+            Objects.requireNonNull(id);
+            return new CommunicationUserIdentifier(id);
+        }
+
+        if (kind == CommunicationIdentifierKind.CALLING_APPLICATION) {
+            Objects.requireNonNull(id);
+            return new CallingApplicationIdentifier(id);
+        }
+
+        if (kind == CommunicationIdentifierKind.PHONE_NUMBER) {
+            Objects.requireNonNull(identifier.getPhoneNumber());
+            Objects.requireNonNull(identifier.getId());
+            return new PhoneNumberIdentifier(identifier.getPhoneNumber()).setId(identifier.getId());
+        }
+
+        if (kind == CommunicationIdentifierKind.MICROSOFT_TEAMS_USER) {
+            Objects.requireNonNull(identifier.getMicrosoftTeamsUserId());
+            Objects.requireNonNull(identifier.getCloudEnvironmentModel());
+            return new MicrosoftTeamsUserIdentifier(identifier.getMicrosoftTeamsUserId(), identifier.isAnonymous())
```

Comment:
We also need `Objects.requireNonNull` on the id and isAnonymous

Reply:
But full id is "Required optional", passing null is assumed ok?